frontend/markdown-pdf/build.js

```diff
@@ -113,77 +157,668 @@ Promise.all([
-                        .newPage()
-                        .then(page =>
-                            page.emulateMedia('screen').then(() => page)
-                        ),
-                ]).then(async ([html, page]) => {
-                    await page.setContent(html);
-                    return page.pdf({
-                        format: 'A4',
-                        margin: {
-                            top: '25mm',
-                            right: '10mm',
-                            bottom: '25mm',
-                            left: '10mm',
-                        },
-                        displayHeaderFooter: true,
-                        headerTemplate: `
-                            <style>${css}</style>
-                            <header>
-                                <img src="data:image/svg+xml;base64,${logoSVG.toString(
-                                    'base64'
-                                )}" class="logo" />
-                                ${filePath.replace(/\.md$/, '')}
-                            </header>`,
-                        footerTemplate: `
-                            <style>${css}</style>
-                            <footer>
-                                <span class="pageNumber"></span> / <span class="totalPages"></span>
-                            </footer>`,
-                    });
-                }),
-            ]).then(([, pdf]) =>
-                fs.writeFile(
-                    path.join(outputDir, filePath.replace('.md', '.pdf')),
-                    pdf
-                )
+                    pdfs.map(pdf =>
+                        fs.writeFile(
+                            path.join(
+                                outputDir,
+                                filePath.replace('.md', '.pdf')
+                            ),
+                            pdf
+                        )
+                    ),
+                    renderEmptyPage(browser, 'A6'), // Generate an empty page we can use in our booklets
+                ])
             )
         )
-    ).finally(() => browser.close())
+    )
+        .then(([[, emptyPage]]) =>
+            Promise.all(
+                // Loop over all distinct markdown files
+                Array.from(languages.keys(), async manual => {
+                    // Check whether the markdown file exists in the root folder
+                    if (fs.existsSync(path.join(inputDir, manual))) {
+                        // Parse Frontmatter for config extraction
+                        const frontmatter = vuepressUtil.parseFrontmatter(
+                            await fs.readFile(
+                                path.join(inputDir, manual),
+                                'utf8'
+                            )
+                        );
+
+                        // Combine all config elements in the frontmatter into 1 dict
+                        var config = {};
+                        if (frontmatter.data.config) {
```

Comment:
Kunnen we hier een iets meer zeggende naam aan geven dan `config`? Bijv. `booklet` oid? Anders kan het net zo goed op root niveau.